website/src/pages/api/account/delete.ts

```diff
@@ -0,0 +1,55 @@
+import { AxiosError } from "axios";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { getToken } from "next-auth/jwt";
+import { createApiClientFromUser } from "src/lib/oasst_client_factory";
+import { createInferenceClient } from "src/lib/oasst_inference_client";
+import prisma from "src/lib/prismadb";
+import { getBackendUserCore } from "src/lib/users";
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const token = await getToken({ req });
+  if (!token) {
+    return res.status(403).end();
+  }
+
+  if (req.method !== "DELETE") {
+    return res.status(400).end();
+  }
+
+  console.log("deleting user", token.sub);
+  try {
+    const backendUserCore = await getBackendUserCore(token.sub);
+    const client = createApiClientFromUser(backendUserCore);
+    await client.delete_account(backendUserCore);
+    console.log(`user ${token.sub} deleted from data backend`);
+  } catch (err) {
+    console.error("could not delete user from data backend", err);
+    return res.status(500).end();
+  }
+
+  try {
+    const client = createInferenceClient(token);
+    await client.delete_account(token.sub);
+    console.log(`user ${token.sub} deleted from inference`);
+  } catch (err) {
+    if (err instanceof AxiosError && err.response.status === 404) {
+      // user does not exist in the inference backend, they have not send any chats
+      // that is okay, we can continue
+      console.log(`user ${token.sub} does not exist on inference`);
+    } else {
+      console.error("could not delete user from inference backend", err);
```

Comment:
I think we need to setup the logger for frontend soon, since console.log is blocking the cpu